tasks/prettier_conformance/src/spec.rs

```diff
@@ -33,39 +43,69 @@ impl SpecParser {
 }
 
 impl VisitMut<'_> for SpecParser {
+    // Some test cases use a variable to store the parsers.
+    //
+    // ```js
+    // const parser = ["babel"];
+    //
+    // runFormatTest(import.meta, parser, {});
+    // runFormatTest(import.meta, parser, { semi: false });
+    // ````
+    fn visit_variable_declarator(&mut self, decl: &mut VariableDeclarator<'_>) {
+        let Some(name) = decl.id.get_identifier() else { return };
+        if !matches!(name.as_str(), "parser" | "parsers") {
+            return;
+        }
+
+        debug_assert!(self.parsers.is_empty(), "`parsers` is already defined");
+        if let Some(Expression::ArrayExpression(arr_expr)) = &decl.init {
+            for el in &arr_expr.elements {
+                if let ArrayExpressionElement::StringLiteral(literal) = el {
+                    self.parsers.push(literal.value.to_string());
+                }
+            }
+        }
+    }
+
+    // The `runFormatTest()` function is used on prettier's test cases.
+    // We need to collect all calls and get the options and parsers.
     fn visit_call_expression(&mut self, expr: &mut CallExpression<'_>) {
         let Some(ident) = expr.callee.get_identifier_reference() else { return };
-        // The `runFormatTest` function used on prettier's test cases. We need to collect all `run_spec` calls
-        // And then parse the arguments to get the options and parsers
-        // Finally we use this information to generate the snapshot tests
         if ident.name != "runFormatTest" {
             return;
         }
+
+        let mut snapshot_options: SnapshotOptions = vec![];
         let mut parsers = vec![];
-        let mut snapshot_options: Vec<(String, String)> = vec![];
         let mut options = PrettierOptions::default();
 
+        // Get parsers
         if let Some(argument) = expr.arguments.get(1) {
             let Some(argument_expr) = argument.as_expression() else {
                 return;
             };
 
+            // If inlined array
             if let Expression::ArrayExpression(arr_expr) = argument_expr {
-                parsers = arr_expr
-                    .elements
-                    .iter()
-                    .filter_map(|el| {
-                        if let ArrayExpressionElement::StringLiteral(literal) = el {
-                            return Some(literal.value.to_string());
-                        }
-                        None
-                    })
-                    .collect::<Vec<String>>();
+                for el in &arr_expr.elements {
+                    if let ArrayExpressionElement::StringLiteral(literal) = el {
+                        parsers.push(literal.value.to_string());
+                    }
+                }
+            }
+            // If variable
+            if let Expression::Identifier(_) = argument_expr {
```

Comment:
In some cases it was written as `runFormatTest(_, parsers)` instead of `runFormatTest(_, ["babel"])`.